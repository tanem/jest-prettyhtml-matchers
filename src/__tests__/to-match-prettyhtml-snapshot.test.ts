import { createPrettyHtmlMatchers } from '..'
import { svg } from './svg'

describe('default global options', () => {
  beforeEach(() => {
    expect.extend(createPrettyHtmlMatchers())
  })

  test('with default local options', () => {
    expect(svg).toMatchPrettyHtmlSnapshot()
  })

  test('with default local options and hint', () => {
    expect(svg).toMatchPrettyHtmlSnapshot('hint')
  })

  test('with custom local options', () => {
    expect(svg).toMatchPrettyHtmlSnapshot({
      singleQuote: true,
    })
  })

  test('with custom local options and hint', () => {
    expect(svg).toMatchPrettyHtmlSnapshot(
      {
        singleQuote: true,
      },
      'hint'
    )
  })
})

describe('custom global options', () => {
  beforeEach(() => {
    expect.extend(createPrettyHtmlMatchers({ singleQuote: true }))
  })

  test('with default local options', () => {
    expect(svg).toMatchPrettyHtmlSnapshot()
  })

  test('with default local options and hint', () => {
    expect(svg).toMatchPrettyHtmlSnapshot('hint')
  })

  test('with custom local options', () => {
    expect(svg).toMatchPrettyHtmlSnapshot({
      singleQuote: false,
      sortAttributes: true,
    })
  })

  test('with custom local options and hint', () => {
    expect(svg).toMatchPrettyHtmlSnapshot(
      {
        singleQuote: false,
        sortAttributes: true,
      },
      'hint'
    )
  })
})
