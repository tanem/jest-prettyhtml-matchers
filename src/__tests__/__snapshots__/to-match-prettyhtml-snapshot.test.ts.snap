// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`custom global options with custom local options 1`] = `
"<svg
  height=\\"64\\"
  viewBox=\\"0 0 64 64\\"
  width=\\"64\\"
  xmlns=\\"http://www.w3.org/2000/svg\\"
>
  <defs>
    <mask height=\\"100\\" id=\\"MaskTest\\" width=\\"100\\" x=\\"0\\" y=\\"0\\">
      <rect height=\\"32\\" style=\\"stroke:none; fill:white\\" width=\\"64\\" x=\\"0\\" y=\\"0\\" />
    </mask>
  </defs>
  <rect fill=\\"plum\\" height=\\"64\\" stroke=\\"none\\" width=\\"64\\" x=\\"1\\" y=\\"1\\" />
  <rect
    fill=\\"gray\\"
    height=\\"64\\"
    mask=\\"url(#MaskTest)\\"
    stroke=\\"none\\"
    width=\\"64\\"
    x=\\"1\\"
    y=\\"1\\"
  />
</svg>
"
`;

exports[`custom global options with custom local options and hint: hint 1`] = `
"<svg
  height=\\"64\\"
  viewBox=\\"0 0 64 64\\"
  width=\\"64\\"
  xmlns=\\"http://www.w3.org/2000/svg\\"
>
  <defs>
    <mask height=\\"100\\" id=\\"MaskTest\\" width=\\"100\\" x=\\"0\\" y=\\"0\\">
      <rect height=\\"32\\" style=\\"stroke:none; fill:white\\" width=\\"64\\" x=\\"0\\" y=\\"0\\" />
    </mask>
  </defs>
  <rect fill=\\"plum\\" height=\\"64\\" stroke=\\"none\\" width=\\"64\\" x=\\"1\\" y=\\"1\\" />
  <rect
    fill=\\"gray\\"
    height=\\"64\\"
    mask=\\"url(#MaskTest)\\"
    stroke=\\"none\\"
    width=\\"64\\"
    x=\\"1\\"
    y=\\"1\\"
  />
</svg>
"
`;

exports[`custom global options with default local options 1`] = `
"<svg
  xmlns='http://www.w3.org/2000/svg'
  width='64'
  height='64'
  viewBox='0 0 64 64'
>
  <defs>
    <mask id='MaskTest' x='0' y='0' width='100' height='100'>
      <rect x='0' y='0' width='64' height='32' style='stroke:none; fill:white' />
    </mask>
  </defs>
  <rect x='1' y='1' width='64' height='64' stroke='none' fill='plum' />
  <rect
    x='1'
    y='1'
    width='64'
    height='64'
    stroke='none'
    fill='gray'
    mask='url(#MaskTest)'
  />
</svg>
"
`;

exports[`custom global options with default local options and hint: hint 1`] = `
"<svg
  xmlns='http://www.w3.org/2000/svg'
  width='64'
  height='64'
  viewBox='0 0 64 64'
>
  <defs>
    <mask id='MaskTest' x='0' y='0' width='100' height='100'>
      <rect x='0' y='0' width='64' height='32' style='stroke:none; fill:white' />
    </mask>
  </defs>
  <rect x='1' y='1' width='64' height='64' stroke='none' fill='plum' />
  <rect
    x='1'
    y='1'
    width='64'
    height='64'
    stroke='none'
    fill='gray'
    mask='url(#MaskTest)'
  />
</svg>
"
`;

exports[`default global options with custom local options 1`] = `
"<svg
  xmlns='http://www.w3.org/2000/svg'
  width='64'
  height='64'
  viewBox='0 0 64 64'
>
  <defs>
    <mask id='MaskTest' x='0' y='0' width='100' height='100'>
      <rect x='0' y='0' width='64' height='32' style='stroke:none; fill:white' />
    </mask>
  </defs>
  <rect x='1' y='1' width='64' height='64' stroke='none' fill='plum' />
  <rect
    x='1'
    y='1'
    width='64'
    height='64'
    stroke='none'
    fill='gray'
    mask='url(#MaskTest)'
  />
</svg>
"
`;

exports[`default global options with custom local options and hint: hint 1`] = `
"<svg
  xmlns='http://www.w3.org/2000/svg'
  width='64'
  height='64'
  viewBox='0 0 64 64'
>
  <defs>
    <mask id='MaskTest' x='0' y='0' width='100' height='100'>
      <rect x='0' y='0' width='64' height='32' style='stroke:none; fill:white' />
    </mask>
  </defs>
  <rect x='1' y='1' width='64' height='64' stroke='none' fill='plum' />
  <rect
    x='1'
    y='1'
    width='64'
    height='64'
    stroke='none'
    fill='gray'
    mask='url(#MaskTest)'
  />
</svg>
"
`;

exports[`default global options with default local options 1`] = `
"<svg
  xmlns=\\"http://www.w3.org/2000/svg\\"
  width=\\"64\\"
  height=\\"64\\"
  viewBox=\\"0 0 64 64\\"
>
  <defs>
    <mask id=\\"MaskTest\\" x=\\"0\\" y=\\"0\\" width=\\"100\\" height=\\"100\\">
      <rect x=\\"0\\" y=\\"0\\" width=\\"64\\" height=\\"32\\" style=\\"stroke:none; fill:white\\" />
    </mask>
  </defs>
  <rect x=\\"1\\" y=\\"1\\" width=\\"64\\" height=\\"64\\" stroke=\\"none\\" fill=\\"plum\\" />
  <rect
    x=\\"1\\"
    y=\\"1\\"
    width=\\"64\\"
    height=\\"64\\"
    stroke=\\"none\\"
    fill=\\"gray\\"
    mask=\\"url(#MaskTest)\\"
  />
</svg>
"
`;

exports[`default global options with default local options and hint: hint 1`] = `
"<svg
  xmlns=\\"http://www.w3.org/2000/svg\\"
  width=\\"64\\"
  height=\\"64\\"
  viewBox=\\"0 0 64 64\\"
>
  <defs>
    <mask id=\\"MaskTest\\" x=\\"0\\" y=\\"0\\" width=\\"100\\" height=\\"100\\">
      <rect x=\\"0\\" y=\\"0\\" width=\\"64\\" height=\\"32\\" style=\\"stroke:none; fill:white\\" />
    </mask>
  </defs>
  <rect x=\\"1\\" y=\\"1\\" width=\\"64\\" height=\\"64\\" stroke=\\"none\\" fill=\\"plum\\" />
  <rect
    x=\\"1\\"
    y=\\"1\\"
    width=\\"64\\"
    height=\\"64\\"
    stroke=\\"none\\"
    fill=\\"gray\\"
    mask=\\"url(#MaskTest)\\"
  />
</svg>
"
`;
